utils/decorators.py

```diff
@@ -32,3 +34,19 @@ def _wrapped_view(request: WSGIRequest, *args: dict, **kwargs: dict) -> Callable
         return _wrapped_view
 
     return decorator
+
+
+def check_manager_permissions(redirect_path: Optional[str] = None) -> Callable:
```

Comment:
This should be 100% doable using existing mixin classes. But real problem is that `ExportAuthorProjectReports` and `AuthorReportProjectView` are using `model = CustomUser` while they should use `model = Project`. The the target project should be returned by view queryset, which shoul be limited by mixin classes.

We basically added those mixins to avoid making extra query for main view object, like you are doing in this decorator. So please try to apply those views to use `model = Project` and apply existing mixin classes. You may have to refactor  `author_report_list.html` template and change 

```
{% regroup object.get_reports_created by date as reports_created_by_date %}
``` 

so instead of `object.get_reports_created` the report list will be already passed in a variable like `reports_list` from view.

Reply:
I understand now, what was the problem. This mixin `get_queryset` was overwritten so we was able to see other report in project even when I wasn't manager of project. I know that the same situations was in other views.

Reply:
Partially yes, also the problem was that you were querying for different model here, but it looks good now.